src/test/java/persistence/study/ReflectionTest.java

```diff
@@ -0,0 +1,157 @@
+
+        // When
+        final List<Method> methods = Arrays.stream(carClass.getDeclaredMethods())
+                .filter(m -> m.getName().startsWith("test"))
+                .collect(Collectors.toList());
+        List<String> actualResults = new ArrayList<>();
+        for (Method method : methods) {
+            actualResults.add((String) method.invoke(car));
+        }
+
+        // Then
+        String expectedTestGetNameResult = String.format("test : %s", name);
+        String expectedTestGetPriceResult = String.format("test : %d", price);
+        assertThat(actualResults).contains(expectedTestGetNameResult, expectedTestGetPriceResult);
+    }
+
+    @DisplayName("@PrintView 애노테이션 메소드 실행")
+    @Test
+    void testAnnotationMethodRun() throws NoSuchMethodException, InvocationTargetException, InstantiationException, IllegalAccessException {
+        // Given
+        final Class<Car> carClass = Car.class;
+        final Car car = carClass.getConstructor().newInstance();
+
+        // When
+        final List<Method> methods = Arrays.stream(carClass.getDeclaredMethods())
+                .filter(m -> m.isAnnotationPresent(PrintView.class))
+                .collect(Collectors.toList());
+
+        // Then
+        for (Method method : methods) {
+            method.invoke(car);
+        }
+    }
+
+    @DisplayName("private field에 값 할당")
+    @Test
+    void privateFieldAccess() throws NoSuchMethodException, InvocationTargetException, InstantiationException, IllegalAccessException {
+        // Given
+        final String name = "simpson";
+        final int price = 100;
+        final Class<Car> carClass = Car.class;
+        final Car car = carClass.getConstructor().newInstance();
+
+        // When
+        final List<Field> fields = Arrays.stream(carClass.getDeclaredFields())
+                .filter(f -> f.getModifiers() == Modifier.PRIVATE)
+                .collect(Collectors.toList());
+
+        for (Field privateField : fields) {
+            privateField.setAccessible(true);
+            changeCarName(name, car, privateField);
+            changeCarPrice(price, car, privateField);
+        }
+
+        // Then
+        assertThat(car.getName()).isEqualTo(name);
+        assertThat(car.getPrice()).isEqualTo(price);
+    }
+
+    @DisplayName("인자를 가진 생성자의 인스턴스 생성")
+    @Test
+    void constructorWithArgs() throws NoSuchMethodException, InvocationTargetException, InstantiationException, IllegalAccessException {
+        // Given
+        final String name = "simpson";
+        final int price = 100;
+        final Class<Car> carClass = Car.class;
+
+        // When
+        final Car car = carClass.getConstructor(String.class, int.class).newInstance(name, price);
+
+        // Then
+        assertThat(car).isEqualTo(new Car(name, 100));
+    }
+
+    private void changeCarPrice(final int price, final Car car, final Field privateField) throws IllegalAccessException {
```

Comment:
```suggestion
    private void changeCarPrice(final int price, final Car car, final Field field) throws IllegalAccessException {
```

`changeCarPrice(price, car, privateField);`
`privateField` 이름은 호출 하는곳에서의 의미이지 않을까요?

주입 인자 `Field`를 사용하는 함수 내부에서는 `private`여부는 중요하지 않는것 같아요. 
private 필드가 아니어도 주입이 가능하므로, 단순히 field라고 명명해도 충분하지 않을까 생각이되네요.
개인적으로는 확장 가능한 설계에서는 인자의 이름 또한 중요하다고 생각해서 관점 공유해봅니다~ 😄

Reply:
아 그렇군요! 감사합니다!